#Robot Frames (CHECK URDF)

isQuadruped: true

base_link: "pelvis"
LFfoot: "wheel_1"
RFfoot: "wheel_2"
LHfoot: "wheel_3"
RHfoot: "wheel_4"
modelname: "/home/master/dp_ws/src/serow/share/urdf/centauro.urdf"

useLegOdom: true

#ROS Topic Names
odom_topic: "/svo/odom0" #only if usePoseUpdate is false and useLegOdom is false
#imu_topic: "/xbotcore/cogimon/imu/imu_link"
imu_topic: "/xbotcore/imu/imu_link"
#joint_state_topic: "/cogimon/joint_states"
joint_state_topic: "/xbotcore/ros_joint_states"
#lfoot_force_torque_topic: "/xbotcore/cogimon/ft/l_leg_ft"
#rfoot_force_torque_topic: "/xbotcore/cogimon/ft/r_leg_ft"
LFfoot_force_torque_topic: "/cartesian/force_estimation/wheel_1"
RFfoot_force_torque_topic: "/cartesian/force_estimation/wheel_2"
LHfoot_force_torque_topic: "/cartesian/force_estimation/wheel_3"
RHfoot_force_torque_topic: "/cartesian/force_estimation/wheel_4"

#Ground-Truth #Not Mandatory - For comparison only
ground_truth: true
ground_truth_odom_topic: "/vrpn_client_node/centauro/pose"
ground_truth_com_topic: "/cogimon/ground_truth/com/odom"
is_in_ds_topic: "/ds"
T_B_GT: [0, 0, 1, 0, 1, 0, 0, 0, 0, 1 , 0, 0, 0, 0 ,0, 1]


comp_with: false
comp_with_odom0_topic: "/svo/odom0"


#If external Support Foot Detection is provided
support_idx_provided: false
support_idx_topic: "/sp"

#To publish relative data
debug_mode: true

#TF from IMU to Body
T_B_A: [1, 0, 0, 0, 0, 1, 0, 0, 0, 0 , 1, 0, 0, 0 ,0, 1]
T_B_G: [1, 0, 0, 0, 0, 1, 0, 0, 0, 0 , 1, 0, 0, 0 ,0, 1]
#TF from F/T to Left Foot
T_FT_LF: [1, 0, 0, 0, 0, 1, 0, 0, 0, 0 , 1 , 0, 0, 0, 0, 1]
T_FT_LH: [1, 0, 0, 0, 0, 1, 0, 0, 0, 0 , 1 , 0, 0, 0, 0, 1]

#TF from F/T to Right Foot
T_FT_RF: [1, 0, 0, 0, 0, 1, 0, 0, 0, 0 , 1 , 0, 0, 0, 0, 1]
T_FT_RH: [1, 0, 0, 0, 0, 1, 0, 0, 0, 0 , 1 , 0, 0, 0, 0, 1]



#TF from External Pose to Body only if useLegOdom is false
T_B_P: [0, 0, 1, 0, -1, 0, 0, 0, 0, -1 ,0 ,0, 0, 0, 0, 1] #SVO without imu

#ROS Topic Frequencies in Hz
imu_topic_freq: 200
joint_topic_freq: 200
fsr_topic_freq: 25  

mass: 70.5957  #robot mass

##Schmitt Trigger - Contact Classifier
LegUpThres: 142 #Schmitt Trigger High Threshold in Newtons (N)
LegLowThres: 42.5 #Schmitt Trigger Low Threshold in Newtons (N)
LosingContact: 20.0
StrikingContact: 800
medianWindow: 10

#Compute Joint Velocities
joint_cutoff_freq: 16
joint_noise_density: 0.025  # rad


useGEM: true
LFforce_sigma: 3.0
LHforce_sigma: 3.0
RFforce_sigma: 3.0
RHforce_sigma: 3.0

probabilisticContactThreshold: 0.95
ContactDetectionWithCOP: true
foot_polygon_xmin: -0.1
foot_polygon_xmax: 0.1
foot_polygon_ymin: -0.05
foot_polygon_ymax: 0.05
LFcop_sigma: 0.005
RFcop_sigma: 0.005
LHcop_sigma: 0.005
RHcop_sigma: 0.005
ContactDetectionWithKinematics: true
VelocityThres: 0.2
LFvnorm_sigma: 0.05
RFvnorm_sigma: 0.05
LHvnorm_sigma: 0.05
RHvnorm_sigma: 0.05
#Mahony Filter for attitude
useMahony: true
Mahony_Kp: 0.001
Mahony_Ki: 0.0
#Madgwick Filter for attitude if useMahony is false
Madgwick_gain: 0.00001




#Leg Odometry Coefficients
Tau0: 1.0
Tau1: 1.0



#Choose Integration Method
#Euler Method for faster execution if false the Runge-Kutta 4 method will be used 
useEuler: true
contact_random_walk: 0.05




#Rigid body Extended Kalman Filter (EKF)
#To estimate:
# 3-D Body Position/Velocity
# 3-D IMU-Biases

#Process Noise STDs
accelerometer_bias_random_walk:  2.4336e-05 #m/s^2/sqrt(s)
gyroscope_bias_random_walk: 1.0e-6 # rad/s/sqrt(s)
accelerometer_noise_density: 0.5 # m/s^2  #Continuous Time
gyroscope_noise_density: 5.0e-02 # rad/s  #Continuous Time

#Measuremets
#Odometry Measurement Noise std:
#LO
leg_odom_position_noise_density: 5.0e-04
leg_odom_orientation_noise_density: 5.0e-02
#VO

odom_orientation_noise_density: 5.0e-02 #4
odom_position_noise_density_x: 3.5e-02 #2.0e-01
odom_position_noise_density_y: 3.5e-02
odom_position_noise_density_z: 1.0



velocity_noise_density_x: 0.035 #0.007
velocity_noise_density_y: 0.035 #0.009
velocity_noise_density_z: 0.035 #0.007







#IMU Initial Biases in the IMU local frame - Biases are added!

bias_ax: 0.35753
bias_ay: 0.00
bias_az: 0.00
bias_gx: -1.24626e-07
bias_gy: -4.49618e-07
bias_gz: 2.82382e-08



#Filter Params
gravity: 9.80665 #gravity constant (m/s^2)


# CoM Extended Kalman Filter (EKF)
# To estimate:
# CoM 3-D Position/Velocity
# 3-D External Forces on the CoM
estimateCoM: false   #Use the CoM EKF

#---------------------------------------------#
#Specify the rest only if estimateCoM is true!!
#---------------------------------------------#

#Process Noise
com_position_random_walk: 1.0e-3 #in m
com_velocity_random_walk: 5.0e-1 #in m/s
external_force_random_walk: 5.0  #in Newtons
#Measurment Noise
com_position_noise_density: 1.0e-03  #CoM position in m
com_acceleration_noise_density: 1.9519 #CoM Acceleration in m/s^2
#Initial errors in External Forces
bias_fx: 0.0  #in Newtons
bias_fy: 0.0
bias_fz: 0.0
#Gyro LPF
useGyroLPF: false
#Cutt-off Frequencies
gyro_cut_off_freq: 10.0  #Only if useGyroLPF is true
#Gyro Moving Average Filter
maWindow: 5 #buffer size, the larger the smoother and delayed the signal, only if useGyroLPF is false
#Filter Params
Ixx : 0.038130318  #torso inertia around x
Iyy : 0.0089760596  #torso inetria around y
Izz : 0.039650845  #torso inertia around z
